lib/Extractor/Solver.cpp

```diff
@@ -111,6 +115,13 @@ class BaseSolver : public Solver {
       }
     }
 
+    if (InferInsts && SMTSolver->supportsModels()) {
+      InstSynthesis IS;
+      EC = IS.synthesize(SMTSolver.get(), BPCs, PCs, LHS, RHS, IC, Timeout);
+      if (EC || RHS)
```

Comment:
Please make sure it is OK to read RHS here (that is, make sure synthesize() has set it to 0 when synthesis failed. I ran into a bug of this kind recently.